helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -656,6 +656,17 @@ public InstanceOperation getInstanceOperation() {
           .build();
     }
 
+    // if instance operation id DISABLE, we override it to ENABLE if instance is enabled
+    if (activeInstanceOperation.getOperation().equals(InstanceConstants.InstanceOperation.DISABLE)) {
+      // it is not likely that instanceEnabled is unset, because when we set operation to disable,
+      // we always set instanceEnabled to false
```

Comment:
Could you please clarify the comments? Does 'InstanceEnabled' refer to 'HELIX_ENABLED'?